docs/implement-your-own.md

```diff
@@ -0,0 +1,92 @@
+# Implement your own
+
+When writing your own Plugin, you need to be aware that the Plugin Client is async first.
+This means that every plugin must be written with Promises. More about this later.
+
+Each plugin must implement the `Http\Client\Plugin\Plugin` interface.
+
+This interface defines the `handleRequest` method that allows to modify behavior of the call:
+
+```php
+/**
+ * Handles the request and returns the response coming from the next callable.
```

Comment:
Oh, you also do something like that (modifying the response) in one example below.